editor/src/messages/tool/common_functionality/overlay_renderer.rs

```diff
@@ -45,8 +45,9 @@ impl OverlayRenderer {
 	}
 
 	pub fn render_subpath_overlays(&mut self, selected_shape_state: &SelectedShapeState, document: &Document, layer_path: Vec<LayerId>, responses: &mut VecDeque<Message>) {
-		let transform = document.generate_transform_relative_to_viewport(&layer_path).ok().unwrap();
-		if let Ok(layer) = document.layer(&layer_path) {
+		let transform = document.generate_transform_relative_to_viewport(&layer_path).ok();
+		let layer = document.layer(&layer_path);
+		if let (Ok(layer), Some(transform)) = (layer, transform) {
```

Comment:
This sort of works around the issue without really solving it. The core issue is that we appear to handle the `DeleteSelectedLayers` message after the `SelectionChanged` message when handling `DeleteSelectedLayers`. This means that the selection is updated before the layer is actually deleted.

Reply:
The solution for this would be to modify the implementation of the `DeleteSelectedLayers` message to ensure that `DeleteLayer` is called before `SelectionChanged`.

Reply:
Thanks, I had a feeling that might be the case.

Is the current solution any better?